### DISCUSSION TITLE
Idea for Editor: Snap to geographic extent of map based on controlled geographic metadata

### DISCUSSION THREAD

**Discussion by itspangler** [allmaps/allmaps#155] [Ideas for Apps: Editor, Viewer, ...]
It would be amazing to add a feature in Allmaps that automatically checks for existing geographic metadata in the map source. e.g., this Sanborn map of Fort Lupton in Colorado contains the following spatial metadata:

- State/Province
   - [Colorado](https://www.loc.gov/search/?fa=location_state:colorado)
- County
   - [Weld County](https://www.loc.gov/search/?fa=location_county:weld+county)
- City
   - [Fort Lupton](https://www.loc.gov/search/?fa=location_city:fort+lupton)

When a map is loaded in the "Georeference" tab of Allmaps Editor, could the application follow this logic?

1. if control points exist, snap to extent of control points
2. if no control points exist, try snapping to highest available resolution of geographic metadata
3. if no geographic metadata exists, default to full view of the world

But because institutions use different controlled vocabularies and metadata schemas, and because this information isn't included in a IIIF manifest by default, perhaps this is too complicated.

**Comment by bertspaan** [comment]
Great idea! Easiest would be to see if the manifest contains a `navPlace` property. Or we could create something that uses named entity recognition that parses the metadata from manifests and then geocodes these? Unfortunately the LOC manifests don't contain any useful geographic metadata that's easy to parse: https://www.loc.gov/item/sanborn00997_001/manifest.json
